Resolution Calculation Broken

```
2014-10-20 15:22:34.026 [clojure-agent-send-off-pool-16] ERROR resolution-q-00 kixi.hecuba.storage.db - Query execution failed: SELECT * FROM partitioned_measurements WHERE device_id = '879edb90f2a7c5bf191f
e6f5fa10a47176c5eced' AND type = 'temperatureRadiant' ORDER BY type desc LIMIT 100; opts: null
clojure.lang.ExceptionInfo: Query execution failed
        at clojure.core$ex_info.invoke(core.clj:4403) ~[clojure-1.6.0.jar:na]
        at qbits.alia$ex__GT_ex_info.invoke(alia.clj:168) ~[na:na]
        at qbits.alia$ex__GT_ex_info.invoke(alia.clj:170) ~[na:na]
        at qbits.alia$execute.invoke(alia.clj:273) ~[na:na]
        at kixi.hecuba.storage.db.CassandraSession._execute(db.clj:55) ~[na:na]
        at kixi.hecuba.storage.db$execute.doInvoke(db.clj:87) [na:na]
        at clojure.lang.RestFn.invoke(RestFn.java:425) [clojure-1.6.0.jar:na]
        at kixi.hecuba.data.calculate$resolution.invoke(calculate.clj:279) [na:na]
        at kixi.hecuba.controller.pipeline$build_pipeline$fn__36570$resolution_q_consumer__36575.invoke(pipeline.clj:307) [na:na]
        at kixi.hecuba.controller.pipeline$build_pipeline$fn__36570.invoke(pipeline.clj:305) [na:na]
        at pipejine.core$consumer$deliver__9263.invoke(core.clj:49) [na:na]
        at pipejine.core$consumer.invoke(core.clj:69) [na:na]
        at pipejine.core$spawn_consumers$fn__9279.invoke(core.clj:91) [na:na]
        at clojure.core$binding_conveyor_fn$fn__4145.invoke(core.clj:1910) [clojure-1.6.0.jar:na]
        at clojure.lang.AFn.call(AFn.java:18) [clojure-1.6.0.jar:na]
        at java.util.concurrent.FutureTask.run(FutureTask.java:262) [na:1.7.0_67]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_67]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_67]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_67]
Caused by: com.datastax.driver.core.exceptions.InvalidQueryException: Partition key part month must be restricted since preceding part is
```

--- a/src/kixi/hecuba/data/calculate.clj
+++ b/src/kixi/hecuba/data/calculate.clj
@@ -12,7 +12,8 @@
             [kixi.hecuba.data.profiles :as profiles]
             [kixi.hecuba.data.misc :as m]
             [kixi.hecuba.storage.db :as db]
-            [qbits.hayt :as hayt]))
+            [qbits.hayt :as hayt]
+            [kixi.hecuba.time :as time]))
 
 ;; Helpers
 
@@ -270,18 +271,23 @@
   Depending on the order of data in the database, it might not always return last 100 of measurements."
   [store item]
   (db/with-session [session (:hecuba-session store)]
-    (let [sensors-to-update (filter #(empty? (:resolution %)) (db/execute session (hayt/select :sensors)))]
+    (let [sensors-to-update (filter #(empty? (:resolution %)) (m/all-sensors store))]
       (doseq [sensor sensors-to-update]
-        (let [measurements (db/execute session (hayt/select :partitioned_measurements
-                                                            (hayt/where [[= :device_id (:device_id sensor)]
-                                                                         [= :type (:type sensor)]])
-                                                            (hayt/order-by [:type :desc])
-                                                            (hayt/limit 100)))
-              resolution  (str (find-resolution measurements))]
-          (db/execute session (hayt/update :sensors
-                                           (hayt/set-columns [[:resolution resolution]])
-                                           (hayt/where [[= :device_id (:device_id sensor)]
-                                                        [= :type (:type sensor)]]))))))))
+        (when (and (:lower_ts sensor) (:upper_ts sensor))
+          (let [months       (time/range->months (:lower_ts sensor) (:upper_ts sensor))
+                measurements (db/execute session (hayt/select :partitioned_measurements
+                                                              (hayt/where [[= :device_id (:device_id sensor)]
+                                                                           [= :type (:type sensor)]
+                                                                           [:in :month months]])
+                                                              (hayt/limit 100)))]
+            (when (seq measurements)
+              (let [calculated-resolution  (str (find-resolution measurements))]
+                (log/debugf "Checking resolution for id: %s type: %s. Calculated resolution is: %s"
+                          (:device_id sensor) (:type sensor) calculated-resolution)
+                (db/execute session (hayt/update :sensors
+                                                 (hayt/set-columns [[:resolution calculated-resolution]])
+                                                 (hayt/where [[= :device_id (:device_id sensor)]
+                                                              [= :type (:type sensor)]])))))))))))
 (defn update-resolution [store sensor resolution]
   (db/with-session [session (:hecuba-session store)]
     (db/execute session (hayt/update :sensors (hayt/set-columns {:resolution (str resolution)})

